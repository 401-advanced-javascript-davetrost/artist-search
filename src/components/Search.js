import React from 'react';
import PropTypes from 'prop-types';

export default function Search({ handleSubmit, handleChange }) {
  return (
    <form onSubmit={handleSubmit}>
      <input type="text" name="artistSearch" placeholder="Artist Name" onChange={handleChange}></input> 
      <button>Search</button>
    </form>
  );
}

Search.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired
};
