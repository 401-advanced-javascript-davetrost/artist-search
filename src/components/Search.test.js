import React from 'react';
import { shallow } from 'enzyme';
import Search from './Search';

describe('App component', () => {
  const clickFunc = jest.fn();
  const typingFunc = jest.fn();

  it('renders App', () => {
    const wrapper = shallow(<Search 
      handleSubmit={clickFunc}
      handleChange={typingFunc}
    />);
    expect(wrapper).toMatchSnapshot();

    wrapper.find('input').simulate('change');
    expect(typingFunc).toHaveBeenCalledTimes(1);
    wrapper.find('input').simulate('change');
    expect(typingFunc).toHaveBeenCalledTimes(2);

    wrapper.find('form').simulate('submit');
    expect(clickFunc).toHaveBeenCalledTimes(1);
  });
});
