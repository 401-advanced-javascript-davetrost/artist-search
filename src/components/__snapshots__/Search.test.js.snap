// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<form
  className="Search"
  onSubmit={[MockFunction]}
>
  <input
    name="artistSearch"
    onChange={[MockFunction]}
    placeholder="Artist Name"
    type="text"
  />
  <button>
    Search
  </button>
</form>
`;
